bedms/train.py

```diff
@@ -0,0 +1,242 @@
+""" This is the training script with which the user can train their own models."""
+
+import logging
+import torch
+from torch import nn
+from torch import optim
+from sklearn.metrics import (
+    precision_score,
+    recall_score,
+    f1_score,
+)
+import yaml
+from .utils_train import (
+    load_from_dir,
+    accumulate_data,
+    training_encoding,
+    data_loader,
+    train_model,
+    plot_learning_curve,
+    model_testing,
+    plot_confusion_matrix,
+    auc_roc_curve,
+)
+from .const import PROJECT_NAME
+from .model import BoWSTModel
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(PROJECT_NAME)
+
+
+class AttrStandardizerTrainer:
+    """
+    This is the training class responsible for
+    managing the training process for the standardizer model.
+    """
+
+    def __init__(self, config: str) -> None:
+        """
+        Initializes the TrainStandardizer object with the given configuration.
+
+        :param str config: Path to the config file which has the training parameters provided by the user.
+        """
+        self.label_encoder = None
+        self.vectorizer = None
+        self.train_loader = None
+        self.val_loader = None
+        self.test_loader = None
+        self.output_size = None
+        self.criterion = None
+        self.train_accuracies = None
+        self.val_accuracies = None
+        self.train_losses = None
+        self.val_losses = None
+        self.model = None
+        self.fpr = None
+        self.tpr = None
+        self.roc_auc = None
+        self.all_labels = None
+        self.all_preds = None
+
+        with open(config, "r") as file:
+            self.config = yaml.safe_load(file)
```

Comment:
Can we validate the config after this? there are a lot of functions that dig into it without really checking all the values are there, so maybe a short `_validate_config` to make sure the config the user provided is good and wont cause issues so they can get immediate feedback